libindex/tempfile_linux.go

```diff
@@ -1,24 +1,75 @@
 package libindex
 
 import (
+	"errors"
 	"fmt"
 	"os"
+	"sync"
 
 	"golang.org/x/sys/unix"
 )
 
+// Holds: dir → error
+var fsChecks sync.Map
+
+// CheckFS returns the memoized result of trying to open a tempfile in the
+// specified directory.
```

Comment:
What is the reason for attempting to use `O_TMPFILE` (and falling back to attempting to use `CreateTempDir()`) and then based on the results actually doing one or the other as opposed to just keeping all the logic in `openTemp` (including the "cached" results). Not sure if that's a good explanation, like why check when the actual opening attempts can serve as the check?

Reply:
This grew out of trying to use `statfs`, so that's probably why. I'll stream-line it.